SocialCareCaseViewerApi/V1/Gateways/ProcessDataGateway.cs

```diff
@@ -51,18 +70,31 @@ public IEnumerable<CareCaseData> GetProcessData(ListCasesRequest request)
             return ResponseFactory.ToResponse(result);
         }
 
-        public IEnumerable<CareCaseData> GetProcessData(long mosaicId, string officerEmail)
+        public IEnumerable<CareCaseData> GetProcessData(long? mosaicId, string officerEmail, string firstName, string lastName, string dateOfBirth, string postcode)
         {
-            var filter = !string.IsNullOrEmpty(officerEmail) ?
-               Builders<BsonDocument>.Filter.Eq("worker_email", officerEmail)
-               : "{$or: [{ mosaic_id: " + mosaicId.ToString() + "}, { mosaic_id: /" + mosaicId.ToString() + "/}]}";
-
-            /*var sFilter = "{ $expr: " +
-                                "{ $regexMatch: " +
-                                    "{ input: { $toString: $mosaic_id }, $regex: /" + mosaicId.ToString() + "/ } " +
-                            "} }";*/
+            var officerEmailFilter = !string.IsNullOrWhiteSpace(officerEmail) ?
+                Builders<BsonDocument>.Filter.Eq("worker_email", officerEmail) : null;
+            var mosaicIdFilter = !string.IsNullOrWhiteSpace(mosaicId.ToString()) ?
```

Comment:
check for null mosaicId before conversion

Reply:
Updated